promise/Promise.ts

```diff
@@ -115,10 +115,22 @@ export function wait<T>(interval:number = 0): Promise<T> {
  * @param condition - function that should return true when condition is met
  * @param interval - milliseconds to wait between checks (default 100). Note that the actual time
  *      will be subject to the minimum delay for `setTimeout()` in the browser.
+ * @param timeout - milliseconds after which the Promise should be rejected (default Infinity).
  */
-export function waitFor(condition: () => boolean, interval: number = 100): Promise<void> {
-    return new Promise(resolver => {
-        const resolveOnMet = () => condition() ? resolver() : setTimeout(resolveOnMet, interval);
+export function waitFor(
+    condition: () => boolean,
+    interval: number = 100,
+    timeout: number = Infinity
```

Comment:
Would prefer timeout of 5 seconds

Reply:
And I think a tighter default interval!  That is one of the benefits of this approach. 50ms would be consistent with the loop utils.